creator-node/src/services/stateMachineManager/monitoring/processStateMonitoringJob.js

```diff
@@ -0,0 +1,276 @@
+    let replicaSetNodesToUserClockStatusesMap
+    try {
+      // Set mapping of replica endpoint to (mapping of wallet to clock value)
+      const clockStatusResp =
+        await retrieveClockStatusesForUsersAcrossReplicaSet(
+          replicaSetNodesToUserWalletsMap
+        )
+      replicaSetNodesToUserClockStatusesMap =
+        clockStatusResp.replicasToUserClockStatusMap
+
+      // Mark peers as unhealthy if they were healthy before but failed to return a clock value
+      unhealthyPeers = new Set([
+        ...unhealthyPeers,
+        ...clockStatusResp.unhealthyPeers
+      ])
+
+      _addToDecisionTree(
+        decisionTree,
+        jobId,
+        'retrieveClockStatusesForUsersAcrossReplicaSet Success'
+      )
+    } catch (e) {
+      _addToDecisionTree(
+        decisionTree,
+        jobId,
+        'retrieveClockStatusesForUsersAcrossReplicaSet Error',
+        { error: e.message }
+      )
+      throw new Error(
+        'processStateMonitoringJob retrieveClockStatusesForUsersAcrossReplicaSet Error'
+      )
+    }
+
+    // Retrieve success metrics for all users syncing to their secondaries
+    let userSecondarySyncMetricsMap = {}
+    try {
+      userSecondarySyncMetricsMap =
+        await computeUserSecondarySyncSuccessRatesMap(nodeUsers)
+      _addToDecisionTree(
+        decisionTree,
+        jobId,
+        'computeUserSecondarySyncSuccessRatesMap Success',
+        {
+          userSecondarySyncMetricsMapLength: Object.keys(
+            userSecondarySyncMetricsMap
+          ).length
+        }
+      )
+    } catch (e) {
+      _addToDecisionTree(
+        decisionTree,
+        jobId,
+        'computeUserSecondarySyncSuccessRatesMap Error',
+        { error: e.message }
+      )
+      throw new Error(
+        'processStateMonitoringJob computeUserSecondarySyncSuccessRatesMap Error'
+      )
+    }
+
+    // Find sync requests that need to be issued and ReplicaSets that need to be updated
+    const { requiredUpdateReplicaSetOps, potentialSyncRequests } =
+      await aggregateReconfigAndPotentialSyncOps(
+        nodeUsers,
+        unhealthyPeers,
+        userSecondarySyncMetricsMap,
+        NodeToSpIdManager.getCNodeEndpointToSpIdMap(),
+        THIS_CNODE_ENDPOINT
+      )
+    _addToDecisionTree(
+      decisionTree,
+      jobId,
+      'Build requiredUpdateReplicaSetOps and potentialSyncRequests arrays',
+      {
+        requiredUpdateReplicaSetOpsLength: requiredUpdateReplicaSetOps.length,
+        potentialSyncRequestsLength: potentialSyncRequests.length
+      }
+    )
+  } catch (e) {
+    logger.info(`processStateMonitoringJob ERROR: ${e.toString()}`)
+    jobFailed = true
+  } finally {
+    _addToDecisionTree(decisionTree, jobId, 'END processStateMachineOperation')
+
+    // Log decision tree
+    _printDecisionTree(decisionTree, jobId)
+  }
+
+  // The next job should start processing where this one ended or loop back around to the first user
```

Comment:
increasing the modulo slicing happens in the job completion event and will be removed soon anyway (after DNs update to support pagination)

Reply:
per other comment, i'm comfortable with removing it now

Reply:
synced offline. will keep for now unless all DNs are upgraded